lib/core/src/Cardano/Wallet.hs

```diff
@@ -288,13 +291,27 @@ data WalletLayer s t = WalletLayer
     , listTransactions
         :: DefineTx t
         => WalletId
-        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
+        -> ExceptT ErrNoSuchWallet IO [TransactionInfo]
         -- ^ List all transactions and metadata from history for a given wallet.
         --
         -- The result is sorted on 'slotId' in descending order. The most recent
         -- transaction comes first.
     }
 
+-- | Full information about a transaction, suitable for presentation to the
+-- user.
+data TransactionInfo = TransactionInfo
+    { txId :: Hash "Tx"
+    -- ^ Transaction ID of this transaction
+    , inputs :: [(TxIn, Maybe TxOut)]
```

Comment:
As said, the `inputs` have to be resolved inputs. We will need to store the result of the coin selection with the metadata in order to accommodate this.